net/disk_cache/disk_cache_backend.cc

```diff
@@ -17,12 +22,293 @@
 #include "net/disk_cache/disk_cache.h"
 #include "net/disk_cache/memory/mem_backend_impl.h"
 #include "net/disk_cache/simple/simple_backend_impl.h"
+#include "net/disk_cache/simple/simple_file_enumerator.h"
+#include "net/disk_cache/simple/simple_util.h"
 
 // Completing partial implementation of disk_cache.h, in place of
 // disk_cache.cc.
 namespace disk_cache {
 
-disk_cache::BackendResult CreateCacheBackend(net::CacheType type,
+namespace {
+
+using FileEnumerator = disk_cache::BackendFileOperations::FileEnumerator;
+
+class TrivialFileEnumerator final : public FileEnumerator {
+ public:
+  using FileEnumerationEntry =
+      disk_cache::BackendFileOperations::FileEnumerationEntry;
+
+  explicit TrivialFileEnumerator(const base::FilePath& path)
+      : enumerator_(path) {}
+  ~TrivialFileEnumerator() override = default;
+
+  absl::optional<FileEnumerationEntry> Next() override {
+    return enumerator_.Next();
+  }
+  bool HasError() const override { return enumerator_.HasError(); }
+
+ private:
+  disk_cache::SimpleFileEnumerator enumerator_;
+};
+
+class UnboundTrivialFileOperations
+    : public disk_cache::UnboundBackendFileOperations {
+ public:
+  std::unique_ptr<disk_cache::BackendFileOperations> Bind(
+      scoped_refptr<base::SequencedTaskRunner> task_runner) override {
+    return std::make_unique<disk_cache::TrivialFileOperations>();
+  }
+};
+
+}
+
+BackendResult::BackendResult() = default;
```

Comment:
Please add a note here that this is copied from net/disk_cache/disk_cache.cc with a hint why it's copied.